File: solutions/python/collatz-conjecture/1/collatz_conjecture.py
def steps(number):
    """
    :param number: int - 
    :return total_steps: int - 
    """

    if not isinstance(number, int) or number <= 0:
        raise ValueError("Only positive integers are allowed")

    total_steps = 0;

    while (number != 1):
        
        if number % 2 == 0:
            number /= 2
            total_steps += 1
        else:
            number *= 3
            number += 1
            total_steps += 1

    return total_steps

File: solutions/javascript/collatz-conjecture/1/collatz-conjecture.js
//
// This is only a SKELETON file for the 'Collatz Conjecture' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const steps = (number) => {
  let counter = 0;
  if (number <= 0) {
    throw 'Only positive numbers are allowed';
  } else {
    while (number > 1) {
      if (number % 2 === 0) {
        number = number / 2;
        counter++;
      } else {
        number = (number * 3);
        number++;
        counter++;
      }
    }
  }
  return counter;
};
